src/cascadia/TerminalSettingsModel/Profile.cpp

```diff
@@ -244,17 +245,36 @@ void Profile::LayerJson(const Json::Value& json)
 }
 
 // Method Description:
+// - Either Returns this profile's background image path, if one is set, expanding
 // - Returns this profile's background image path, if one is set, expanding
 //   any environment variables in the path, if there are any.
+// - Or if "DesktopWallpaper" is set, then gets the path to the desktops wallpaper.
 // Return Value:
-// - This profile's expanded background image path / the empty string.
+// - This profile's expanded background image path / desktops's wallpaper path /the empty string.
 winrt::hstring Profile::ExpandedBackgroundImagePath() const
 {
     if (_BackgroundImagePath.empty())
     {
         return _BackgroundImagePath;
     }
-    return winrt::hstring{ wil::ExpandEnvironmentStringsW<std::wstring>(_BackgroundImagePath.c_str()) };
+    // checks if the user would like to copy their desktop wallpaper
+    // if so, replaces the path with the desktop wallpaper's path
+    else if (_BackgroundImagePath == to_hstring(DesktopWallpaperEnum))
+    {
+        WCHAR desktopWallpaper[MAX_PATH];

```

Comment:
The documentation for `SystemParametersInfo` says:

> The returned string will not exceed MAX_PATH characters.

But I'm not positive that is up to date in the post-`MAX_PATH` world of 2020. The API certainly doesn't seem to report a useful error if the actual path is longer than MAX_PATH, something that we could use to resize the buffer to the correct number of characters. 

I'm gonna say that this is okay for now, and let's keep our eyes on it.

Reply:
Would making a desktopWallpaper[MAX_INT] be valuable. We would pass the SPI string to it, then verify the string is not longer than MAX_PATH. If so, we could return an error indicating so. (Would be a good failsafe if the SPI API changes.

Reply:
Well, reserving MAX_INT `wchar_t`s is probably overkill, for an API that I'm assuming only ever writes `MAX_PATH` `wchar_t`s anyways. I'd be _shocked_ if this API changes, without doing something like `SPI_GETDESKWALLPAPEREX`